src/material-experimental/mdc-slide-toggle/slide-toggle.ts

```diff
@@ -142,16 +145,20 @@ export class MatSlideToggle implements ControlValueAccessor, AfterViewInit, OnDe
   /** Whether the label should appear after or before the slide-toggle. Defaults to 'after'. */
   @Input() labelPosition: 'before' | 'after' = 'after';
 
-  /** Used to set the aria-label attribute on the underlying input element. */
+  /** Used to set the aria-label attribute on the underlying button element. */
   @Input('aria-label') ariaLabel: string | null = null;
 
-  /** Used to set the aria-labelledby attribute on the underlying input element. */
+  /** Used to set the aria-labelledby attribute on the underlying button element. */
   @Input('aria-labelledby') ariaLabelledby: string | null = null;
 
-  /** Used to set the aria-describedby attribute on the underlying input element. */
+  /** Used to set the aria-describedby attribute on the underlying button element. */
   @Input('aria-describedby') ariaDescribedby: string;
 
-  /** Whether the slide-toggle is required. */
+  /**
+   * Whether the slide-toggle is required.
+   * @deprecated No longer being used. Will be removed.
```

Comment:
Something you can do if you want to keep form properties and `<form>` support is add a hidden `<input type="checkbox">` in to DOM outside the button and sync it.

```html
<div class="mat-mdc-switch">
  <button
    class="mdc-switch"
    role="switch"
    [class.mdc-switch--selected]="checked"
    [class.mdc-switch--unselected]="!checked"
    [disabled]="disabled"
    [attr.id]="buttonId"
    [attr.name]="name">
    <!-- ... -->
  </button>

  <!-- Hidden input for <form> support -->
  <input 
    type="checkbox"
    style="display: none;" 
    [attr.required]="required"
    [attr.name]="name"
    [ngModel]="checked">
</div>
```

Reply:
That's an interesting approach, but I'd rather avoid it since it adds more complexity to the component. I'll un-deprecate the `required` input since @jelbourn mentioned above that it is supported in ARIA 1.2. We can re-add the `aria-required` later on.